### PR TITLE
feat(grid): add usage page

### LINKED ISSUE #3619: Grid docs: develop Usage page
https://www.figma.com/file/ROcIS7OFtP42sXqH7VODhV/Grid_docs?type=design&node-id=0-1&mode=design&t=wIgTHafnC1lphmzv-0

- [x] Create new Usage page
- [x] Remove Implementation page
- [x] Redirect Implementation url to Usage

---

From @jeanservaas [#13908](https://github.com/carbon-design-system/carbon/issues/13908)

## Figma specs

https://github.com/orgs/carbon-design-system/projects/39/views/28?pane=issue&itemId=23535662

Here are the images. There is ONE image that Mike would like me to make into an animated GIF...but I'm not going to let that hold this up. I can work on that on the side, if it's done in time cool. If not, we'll get to it later.

## PNGs

[Final_grid_images.zip](https://github.com/carbon-design-system/carbon/files/12458495/Final_grid_images.zip)

Also there is one video that we're pulling off the current Grid Implementation tab, so that's not included in this file:

Image

### PR DESCRIPTION
**Pull request by alisonjoseph**
Closes #3619 

#### Changelog

**New**

- 2x Grid Usage page
- Add redirect from Implementation to Usage

**Removed**

- 2x Grid Implementation page
  - related files/images

### CODE REVIEW

File: src/pages/guidelines/2x-grid/usage.mdx [hunk 1]
```diff
@@ -0,0 +1,682 @@
...
+  It’s fundamental to everything we design.
+tabs: ['Overview', 'Usage', 'Code']
+---
+
+import { IbmSecurity } from '@carbon/icons-react';
+import { gridTable } from './Grid.module.scss';
+
+<PageDescription>
+
+You’ve learned the grid fundamentals, now let’s put them togther. A good layout
```
Comment (tw15egan) [inline src/pages/guidelines/2x-grid/usage.mdx:14]:
```suggestion
You’ve learned the grid fundamentals, now let’s put them together. A good layout
```

File: src/pages/guidelines/2x-grid/usage.mdx [hunk 2]
```diff
@@ -0,0 +1,682 @@
...
+
+Layout modules are coded combinations of components, spacing tokens, icons, and
+type styles packaged as a single grid-aware, unit. The layout-module framework
+enables designers to build page layouts with greater rigor, quality, and
+consistency. They are designed to work in tandem with other modules and Carbon
+components so that they can be quickly arranged into layouts with the confidence
+that they are consistent with all of our other pages.
+
+These modules along with the overarching page scaffolding above, provide a great
+example for how product designers should mix and match repeatable layout
```
Comment (tw15egan) [inline src/pages/guidelines/2x-grid/usage.mdx:144]:
```suggestion
example of how product designers should mix and match repeatable layout
```

File: src/pages/guidelines/2x-grid/usage.mdx [hunk 3]
```diff
@@ -0,0 +1,682 @@
...
+[grid influencer content](https://pages.github.ibm.com/cdai-design/pal/patterns/grid-influencers/usage/)
+originated on the Carbon for IBM Products PAL, but we wanted to bring the
+relevant portions of it here too to reach the maximum audience.
+
+### Usage
+
+There are two main use cases where the grid may be influenced by content: left
+navigation and slide-in side panels.
+
+#### Left hand navigation
```
Comment (tw15egan) [inline src/pages/guidelines/2x-grid/usage.mdx:276]:
```suggestion
#### Left-hand navigation
```

File: src/pages/guidelines/2x-grid/usage.mdx [hunk 4]
```diff
@@ -0,0 +1,682 @@
...
+such as hierarchy, scale, proportion, contrast, harmony, rhythm, repetition, and
+many more.
+
+</PageDescription>
+
+<InlineNotification>
+
+Note: Although Carbon is open source, much of our Pattern Asset Library (PAL)
+content is gated—so many of the links below can only be accessed by IBM
+employees.
```
Comment (tay1orjones) [inline src/pages/guidelines/2x-grid/usage.mdx:25]: This is a bummer. I really want to put together some layout samples live with code that we can share publicly.

File: src/pages/guidelines/2x-grid/usage.mdx [hunk 5]
```diff
@@ -0,0 +1,681 @@
...
+  </Column>
+</Row>
+
+### Style models in IBM.com (editorial)
+
+This difference in grid ripples through everything on page. Below is a visual
+overview of how the IBM.com masthead and footer will behave above max
+breakpoints in the appropriate style models. Notice that marketing content on
+IBM.com uses the Editorial style model and that support documentation on IBM.com
+uses the the Product and docs style model.
```
Comment (tw15egan) [inline src/pages/guidelines/2x-grid/usage.mdx:407]:
```suggestion
uses the Product and docs style model.
```

File: src/pages/guidelines/2x-grid/usage.mdx [hunk 6]
```diff
@@ -0,0 +1,681 @@
...
+</Row>
+
+### Style models in IBM Software (product)
+
+Product and editorial experiences overlap slightly in that they both can use the
+product & docs style model. However, only software UI uses the
+[grid influencer](#grid-influencers) variant of the Carbon UI Shell.
+
+IBM Products can also occasionally use the High-density interface model in order
+to maximum the use of screen space. By far the most prominent example of this to
```
Comment (tw15egan) [inline src/pages/guidelines/2x-grid/usage.mdx:449]:
```suggestion
to maximize the use of screen space. By far the most prominent example of this to
```

File: src/pages/guidelines/2x-grid/usage.mdx [hunk 7]
```diff
@@ -0,0 +1,681 @@
...
+
+  </Column>
+</Row>
+
+## Gutter modes
+
+Carbon’s Figma templates always have a 32px gutter by default, regardless of
+breakpoint. The default gutter width is 32px because type, within or without
+containers, never has less than a 32px gutter. For containers and components,
+however, there are three different gutter structu res: wide (default), narrow
```
Comment (tw15egan) [inline src/pages/guidelines/2x-grid/usage.mdx:478]:
```suggestion
however, there are three different gutter structures: wide (default), narrow
```

File: src/pages/guidelines/2x-grid/usage.mdx [hunk 8]
```diff
@@ -0,0 +1,681 @@
...
+| ---------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Editorial model              | This default style model is the bread and butter of marketing pages, which represents the majority of IBM.com. The grid in this model centers content for most comfortable browsing, and allows for a variety of expressive layouts. Some low information density product screens, such as the Checkout application, can use this model too. |
+| Product and docs model       | For long form content with deep organizational hierarchy, use the product and docs model. It anticipates a left hand navigational panel, and keeps content within a maximum width. Examples of documentation include most of IBM product UI, IBM Docs, Cloud Docs, and this site documenting the design system guidance.                     |
```
Comment (tw15egan) [inline src/pages/guidelines/2x-grid/usage.mdx:334]:
```suggestion
| Product and docs model       | For long-form content with deep organizational hierarchy, use the product and docs model. It anticipates a left-hand navigational panel, and keeps content within a maximum width. Examples of documentation include most of IBM product UI, IBM Docs, Cloud Docs, and this site documenting the design system guidance.                     |
```

File: src/pages/guidelines/2x-grid/usage.mdx [hunk 9]
```diff
@@ -0,0 +1,682 @@
...
+modules that are used widely across products. For more detailed documentation
+about structural layout modules, check out
+[IBM Product Layouts](https://pages.github.ibm.com/cdai-design/pal/layouts/library/)
+on the Carbon for IBM Products PAL.
+
+![Grid scaffolding illustration](images/usage/01_basic_scaffolding.png)
+
+<Caption>
+  Although gutter width depends on the product, here are some common examples of
+  how product designers use a consistent leadspace size with the 2x Grid to
```
Comment (tay1orjones) [inline src/pages/guidelines/2x-grid/usage.mdx:100]: Is "leadspace" a unique IBM term? I'm not able to find a solid definition for it. I personally don't know what it refers to. Maybe the page heading? If so, would "page heading" be a more universal way to describe this?

File: src/pages/guidelines/2x-grid/usage.mdx [hunk 10]
```diff
@@ -0,0 +1,681 @@
...
+Fixed components with labels such as input fields and dropdowns MUST use the
+wide gutter mode. So if you’re laying out form fields, this is the gutter mode
+to use. One overarching rule, (applicable to each gutter mode) is that type
+never hangs into the gutter. So we don’t want to push label type into the gutter
+in order to get field type to align to the columns.
+
+![](images/usage/27_wide_gutter_mode_example.png)
+
+<Caption>
+  Note: this UI example also include a left hand navigation panel in
```
Comment (tw15egan) [inline src/pages/guidelines/2x-grid/usage.mdx:540]:
```suggestion
  Note: this UI example also includes a left-hand navigation panel in
```

**Review by tw15egan — COMMENTED** [review]
A few content tweaks

File: src/pages/guidelines/2x-grid/usage.mdx [hunk 11]
```diff
@@ -0,0 +1,682 @@
...
+[mini unit grid](/guidelines/2x-grid/overview/#mini-unit)) need to be used
+rather than fluid columns.
+
+![denser layout example](images/usage/03_denser_layout_UI.png)
+
+### Layout modules and micro layouts
+
+Although Carbon React does not have coded layout modules in its library, the
+[]Carbon for IBM Products
+library](https://v1-ibm-products.carbondesignsystem.com/?path=/story/overview-getting-started--page)
```
Comment (tay1orjones) [inline src/pages/guidelines/2x-grid/usage.mdx]: This link is broke

```suggestion
Although Carbon React does not have coded layout modules in its library, the
[Carbon for IBM Products
library](https://v1-ibm-products.carbondesignsystem.com/?path=/story/overview-getting-started--page)
```

**Comment by alisonjoseph** [comment]
Thanks for taking the time to proof-read @tw15egan I just copy/pasted the content from Figma without reading all of it. 😅

**Review by tay1orjones — APPROVED** [review]
Looks great! I really like how the usage content separates editorial vs product/non-editorial layouts. Just a couple other minor comments.

**Review by tw15egan — APPROVED** [review]

**Review by andreancardona — APPROVED** [review]
🔥

**Comment by jeanservaas** [comment]
Aside from the copy changes I just committed @alisonjoseph the only visual thing I noticed was that the Do/Don't image here gets too close to the check mark/x mark in the large breakpoint...

It looks like these svgs are floating in the Do/Don't component, is there a way to add a slight bit more padding like 8px or 16px max?

image

**Comment by alisonjoseph** [comment]
@jeanservaas the padding will need to be added to the images

**Comment by jeanservaas** [comment]
OK @alisonjoseph let's try this!

[grid_img_updates.zip](https://github.com/carbon-design-system/carbon-website/files/12562404/grid_img_updates.zip)

**Comment by jeanservaas** [comment]
Still maybe not enough padding, or I don't see an improvement... let's see what these do.

[grid_img_updates.zip](https://github.com/carbon-design-system/carbon-website/files/12563216/grid_img_updates.zip)

**Comment by alisonjoseph** [comment]
@jeanservaas 

Screenshot 2023-09-11 at 9 05 57 AM

**Review by jeanservaas — APPROVED** [review]

**Pull request merged**